Sources/Common/DataModel/Box/index.js

```diff
@@ -118,13 +120,11 @@ function vtkBox(publicAPI, model) {
 
   publicAPI.getBounds = () => model.bbox.getBounds();
 
-  publicAPI.evaluateFunction = (...x) => {
-    let point = [].concat(x);
-    if (Array.isArray(x[0])) {
-      point = x[0];
-    }
-
-    if (point.length !== 3) {
+  publicAPI.evaluateFunction = (xyz) => {
```

Comment:
so now we can't use that function like `evaluateFunction (1,2,3)` and `evaluateFunction ([1,2,3])`.

Reply:
- `evaluateFunction ([1,2,3])`: yes, that's the only thing it can, just like for Cone, Cylinder, Sphere, ImplicitBoolean..
- `evaluateFunction (1,2,3)`: no, that's the point: doing the following is super slow:
```js
publicAPI.evaluateFunction = (...x) => {
  let point = [].concat(x);
  if (Array.isArray(x[0])) {
    point = x[0];
  }
```

Reply:
The way it was handle was slow. But that does not mean it has to be slow for handling both use case.

Reply:
Just FYI, any setArray macro will support both method profile and having a limitation on that will be bad as you will never know which one is valid.

Reply:
try that and let me know if that's slow

```
publicAPI.evaluateFunction = (x, y, z) => {
  const point = y === undefined ? x : [x,y,z];

Reply:
if you have a better suggestion please let me know. It's better to support one fast and warn that the second is not supported (so users just update their code), than supporting both and having one silently slow.

Reply:
You can add warning for user that it pick the slow path via an extra

```
if (point !== x) {
   vtkDebug('slow pass');
}